### PR TITLE
[INJIMOB-2415] show credential type in mini view instead of fullName

### PR DESCRIPTION
**Pull request by KiruthikaJeyashankar**
## Description

> Dispaly credential type in mini view instead of fullName

Previously, it was showing hardcoded fullName in mini view, which is now replaced with showing the credential type as per configured issuer metadata

## Issue ticket number and link

> [INJIMOB-2415](https://mosip.atlassian.net/browse/INJIMOB-2415&#41)

## Screenshots

1. Issuer metadata exposes credential type 
image

2. Issuer metadata does not expose display so show fallback

The fallback for displaying credential type when there is no display property exposed in issuer meta data is as follows

- ldp_vc → show value as per credential_definition.type available in issuer metadata
- mso_mdoc / others → show “Identity Card“

image

[INJIMOB-2415]: https://mosip.atlassian.net/browse/INJIMOB-2415?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by swatigoel — CHANGES_REQUESTED** [review]

File: components/VC/common/VCUtils.tsx [hunk 1]
```diff
@@ -248,30 +258,31 @@ export const getMosipLogo = () => {
...
- * supportedCredential's wellknown is passed from getActivityText after fresh download
+ * supportedCredential's wellknown is passed from getActivityText after fresh download and viewing of card's mini view flow
  * & all other consumers pass whole well known response of issuer
  */
-export const getIdType = (
+export const getCredentialType = (
   wellknown: CredentialTypes | IssuerWellknownResponse,
   credentialConfigurationId: string | undefined = undefined,
 ): string => {
+  const defaultCredentialType = i18n.t('VcDetails:nationalCard');
```
[the remark this answers is not in the document]
Comment (KiruthikaJeyashankar) [reply components/VC/common/VCUtils.tsx:271]: Given only ldp_vcs have credential_definition.type, can we populate fallback credential type as per VC formats?

File: components/VC/common/VCUtils.tsx [hunk 2]
```diff
@@ -248,30 +258,31 @@ export const getMosipLogo = () => {
...
- * @returns id Type translations (Eg - National ID)
+ * @param credentialConfigurationId
+ * @returns credential type translations (Eg - National ID)
  *
- * supportedCredential's wellknown is passed from getActivityText after fresh download
+ * supportedCredential's wellknown is passed from getActivityText after fresh download and viewing of card's mini view flow
  * & all other consumers pass whole well known response of issuer
  */
-export const getIdType = (
+export const getCredentialType = (
```
[the remark this answers is not in the document]
Comment (KiruthikaJeyashankar) [reply components/VC/common/VCUtils.tsx:267]: **On #2** As per draft 13 open4vci spec, 
`name: OPTIONAL. String value of a display name for the Credential Issuer.`
Ref- https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0-13.html#name-oauth-20-authorization-serv
Given this is the case, are we mandating name for this wallet app?

**On #3** Given only ldp_vcs have credential_definition.type, can we populate fallback credential type as per VC formats?

**On #4** locale filtering is happening where fallback is set to first element of display array, will change it to accept english locale

File: components/VC/common/VCUtils.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (KiruthikaJeyashankar) [reply components/VC/common/VCUtils.tsx:271]: **Refactor to be made** as per discussion
1. name is mandatory in display of credential_configurations_supported
2. accept only whole wellknown as inputparam for function
3. display? get name : credential_definition.type // get language specifics before getting any data from display

File: components/VC/common/VCUtils.tsx [hunk 3]
```diff
@@ -248,65 +257,106 @@ export const getMosipLogo = () => {
 /**
  *
  * @param wellknown (either supportedCredential's wellknown or whole well known response of issuer)
- * @param idType
- * @returns id Type translations (Eg - National ID)
+ * @param credentialConfigurationId
+ * @returns credential type translations (Eg - National ID)
  *
- * supportedCredential's wellknown is passed from getActivityText after fresh download
+ * supportedCredential's wellknown is passed from getActivityText after fresh download and viewing of card's mini view flow
```
Comment (KiruthikaJeyashankar) [inline components/VC/common/VCUtils.tsx:263]: Can we update this comment with caller flows - mini view & detail view or remove this point?

File: components/VC/common/VCUtils.tsx [hunk 4]
```diff
@@ -248,65 +257,106 @@ export const getMosipLogo = () => {
...
+      }),
+    );
+    return getLocalizedField(idTypeObj);
+  }
+  if (supportedCredentialsWellknown.format === VCFormat.ldp_vc) {
+    const types = supportedCredentialsWellknown.credential_definition
+      .type as string[];
+    return types[1];
+  } else {
+    return i18n.t('VcDetails:nationalCard');
```
Comment (KiruthikaJeyashankar) [inline components/VC/common/VCUtils.tsx:283]: fallback scenario for mso_mdoc = "Identity Card"
https://github.com/mosip/inji-wallet/pull/1705#discussion_r1861512414

File: components/VC/common/VCUtils.tsx [hunk 5]
```diff
@@ -148,18 +146,29 @@ export const getFieldName = (
 };
 
 export const getBackgroundColour = (wellknown: any) => {
+  const defaultBackgroundColor = Theme.Colors.textValue;
+
   return {
     backgroundColor:
-      wellknown?.display[0]?.background_color ?? Theme.Colors.textValue,
+      (wellknown?.display?.length
+        ? wellknown.display[0]?.background_color
```
Comment (swatigoel) [inline components/VC/common/VCUtils.tsx:154]: we don't have to pick 0th element. It has to be filtered based on locale

File: components/VC/common/VCUtils.tsx [hunk 6]
```diff
@@ -148,18 +146,29 @@ export const getFieldName = (
...
+        : null) ?? defaultBackgroundColor,
   };
 };
 
 export const getBackgroundImage = (wellknown: any, defaultBackground: any) => {
-  return wellknown?.display[0]?.background_image ?? defaultBackground;
+  return (
+    (wellknown?.display?.length
+      ? wellknown.display[0]?.background_image
```
Comment (swatigoel) [inline components/VC/common/VCUtils.tsx:162]: we don't have to pick 0th element. It has to be filtered based on locale

File: components/VC/common/VCUtils.tsx [hunk 7]
```diff
@@ -148,18 +146,29 @@ export const getFieldName = (
...
+      : null) ?? defaultBackground
+  );
 };
 
 export const getTextColor = (wellknown: any, defaultColor: string) => {
-  return wellknown?.display[0]?.text_color ?? defaultColor;
+  return (
+    (wellknown?.display?.length ? wellknown.display[0]?.text_color : null) ??
```
Comment (swatigoel) [inline components/VC/common/VCUtils.tsx:169]: we don't have to pick 0th element. It has to be filtered based on locale

File: components/VC/common/VCUtils.tsx [hunk 8]
```diff
@@ -248,65 +257,104 @@ export const getMosipLogo = () => {
...
+    if(!!!supportedCredentialsWellknown['display']){
+      if (supportedCredentialsWellknown.format === VCFormat.ldp_vc) {
+        const credentialDefinition = supportedCredentialsWellknown['credential_definition']
+        let type = credentialDefinition['type'] as Array<string>
+        return type[1];
+      }
+      else {
+        return i18n.t('VcDetails:nationalCard'); // update this to IdentityCard
+      }
+    }
```
Comment (swatigoel) [inline components/VC/common/VCUtils.tsx:320]: why this commented code?

File: components/VC/common/VCUtils.tsx [hunk 9]
```diff
@@ -248,65 +257,104 @@ export const getMosipLogo = () => {
...
+export const getDisplayId1 = (
+  wellknown: IssuerWellknownResponse | CredentialTypes,
+  credentialConfigurationId?: string,
+): string => {
+  let supportedCredentials;
+
+  if (!!wellknown['credential_configurations_supported']) {
+    if (credentialConfigurationId === undefined) {
+      console.error('credentialConfigurationId not available for fetching the Credential type');
+      throw new Error(`Invalid credentialConfigurationId - ${credentialConfigurationId} passed`);
```
Comment (swatigoel) [inline components/VC/common/VCUtils.tsx:342]: why this commented code

**Review by swatigoel — CHANGES_REQUESTED** [review]

**Review by swatigoel — APPROVED** [review]

**Pull request merged**